core/rawdb/database.go

```diff
@@ -301,9 +303,99 @@ func NewLevelDBDatabase(file string, cache int, handles int, namespace string, r
 	if err != nil {
 		return nil, err
 	}
+	log.Info("using LevelDB as the backing database")
 	return NewDatabase(db), nil
 }
 
+// NewPebbleDBDatabase creates a persistent key-value database without a freezer
+// moving immutable chain segments into cold storage.
+func NewPebbleDBDatabase(file string, cache int, handles int, namespace string, readonly bool) (ethdb.Database, error) {
+	db, err := pebble.New(file, cache, handles, namespace, readonly)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("using Pebble as the backing database")
+	return NewDatabase(db), nil
+}
+
+type dbType int
+
+const (
+	Nonexistent dbType = iota
+	Pebble
+	LevelDb
+)
+
+func hasPreexistingDb(path string) dbType {
+	_, err := os.Stat(path + "/CURRENT")
+	if os.IsNotExist(err) || err != nil {
+		return Nonexistent
+	}
```

Comment:
Hm. If error is `IsNotExist`-type, then it will always be non-nil, right? So the `err != nil` should be sufficient. Otherwise it's like `if a > 15 || a > 10`

Reply:
yeah :man_facepalming: